### PR TITLE
Add build number as output variable

### PR DESCRIPTION
**Pull request by jahsu-MSFT**
https://github.com/microsoft/azure-pipelines-tasks/issues/11784

### CODE REVIEW

File: Tasks/DownloadPipelineArtifactV2/task.json [hunk 1]
```diff
@@ -243,5 +243,11 @@
             "target": "Agent.Plugins.PipelineArtifact.DownloadPipelineArtifactTaskV2_0_0, Agent.Plugins"
         }
     },
-    "messages": {}
+    "messages": {},
+    "outputVariables": [
+        {
+            "name": "BuildNumber",
```
Comment (avivanoff) [inline Tasks/DownloadPipelineArtifactV2/task.json]: The name is confusing. Should be PipelineId.

File: Tasks/DownloadPipelineArtifactV2/task.json [hunk 2]
```diff
@@ -243,5 +243,11 @@
...
+            "description": "Stores the build number of the pipeline artifact source"
```
Comment (avivanoff) [inline Tasks/DownloadPipelineArtifactV2/task.json]: The description should say "Stores the id of the pipeline artifact source".

File: Tasks/DownloadPipelineArtifactV2/task.json (hunk 1 above)
Comment (jahsu-MSFT) [reply Tasks/DownloadPipelineArtifactV2/task.json]: Naming it "BuildNumber" so it's consistent with build artifacts tasks

**Comment by jahsu-MSFT** [comment]
@avivanoff That will need to wait until the agent is released. We will deploy agent first and then the task.

**Comment by fadnavistanmay** [comment]
New PR - https://github.com/microsoft/azure-pipelines-tasks/pull/12545 - closing this one.

**Pull request closed without merging**